spec/lib/phobos/producer_spec.rb

```diff
@@ -280,6 +331,28 @@ class TestProducer1
     end
   end
 
+  describe '.sync_producer_shutdown' do
+    let(:producer) { double('Kafka::RegularProducer', produce: true, deliver_messages: true) }
+    let(:kafka_client) { double('Kafka::Client', producer: producer, close: true) }
+
+    it 'calls shutdown in the configured client and cleans up producer' do
+      original_hash = Phobos.config.producer_hash
```

Comment:
consider doing this in a let

`let(:config) { Phobos.config.producer_hash.merge(:persistent_connections => true) }`

and then

`allow(Phobos.config).to receive(:producer_hash).and_return(config)`